scripts/request_flooder.py

```diff
@@ -41,45 +43,72 @@ def parse_args(argv):
                         required=True,
                         metavar='RATE',
                         type=int,
-                        help='The desired request rate in req/5min. Note: the '
-                             'actual request rate can end up being a little '
-                             'slower than the desired rate, more so at higher '
-                             'rates.')
-    parser.add_argument('--time',
-                        metavar='TIME',
+                        help='The desired request rate, or number of requests '
+                             'per DENOMINATOR. Note: the actual request rate '
+                             'can end up being a little slower than the '
+                             'rate specified, more so at higher rates.')
+    parser.add_argument('--per',
+                        metavar='DENOMINATOR',
+                        type=int,
+                        default='300',
+                        help='The DENOMINATOR of the rate in seconds.')
+    parser.add_argument('--duration',
+                        metavar='SECONDS',
                         type=int,
                         default='300',
-                        help='The length of the test in seconds')
+                        help='Total duration of the test in seconds.')
+    parser.add_argument('--log-headers',
+                        default=False,
+                        action='store_true',
+                        help='Include response headers in log output')
     args = parser.parse_args(argv)
-    require(args.url.startswith('http'))
-    require(1 <= args.rate <= 3000,
-            'Request rate must be between 1 and 3000')
-    require(1 < args.time <= 3600,
-            'Request time must be between 1 and 3600')
+    args.method = args.method.upper()
+    assert args.method in ['HEAD', 'GET', 'PUT'], R(
+        'Invalid request method', args.method)
+    assert args.url.startswith('http')
+    assert args.rate / args.per >= 1 / 300, R(
+        'Rate must be at least 1 request per 300 seconds')
+    assert args.rate / args.per <= 10, R(
+        'Rate must be no more than 10 request per second')
+    assert 1 < args.duration <= 3600, R(
+        'Total duration must be between 1 and 3600 seconds')
     return args
 
 
-def head_url(url: str) -> int:
-    response = http.request('HEAD', url)
-    return response.status
+def request_url(method: str, url: str, log_headers: bool) -> int:
+    log.info('Making %s request to %r', method, url)
+    start_time = time.time()
+    # We use requests instead of azul.http.http_client() since we don't want
+    # requests to be retried or delayed due to a 429 response.
```

Comment:
I wouldn't think you need to explain this, scripts are somewhat isolated in our codebase. I may be wrong.